src/sqlacodegen/generators.py

```diff
@@ -1201,18 +1202,28 @@ def render_column_attribute(self, column_attr: ColumnAttribute) -> str:
         column = column_attr.column
         rendered_column = self.render_column(column, column_attr.name != column.name)
 
-        try:
-            python_type = column.type.python_type
+        def recursively_get_col_type(column_type: TypeEngine[Any]) -> str:
+            if column_type.python_type == list:
+                self.add_literal_import("typing", "List")
+                # Dimensions is how postgres handles matrices
+                # See: https://docs.sqlalchemy.org/en/13/dialects/postgresql.html#sqlalchemy.dialects.postgresql.ARRAY
+                dim = getattr(column_type, "dimensions", None) or 1
+
+                return f"{'List[' * dim}{recursively_get_col_type(column_type.item_type)}{']' * dim}"
```

Comment:
As you suggested, and as python 3.9 is now the minimal supported version, we can indeed get rid of `typing.List` and replace it with the standard `list`

Reply:
10x